lms/djangoapps/courseware/course_tools.py

```diff
@@ -91,8 +91,12 @@ def analytics_id(cls):
     @classmethod
     def is_enabled(cls, request, course_key):
         """
-        Show this link for courses where financial assistance is available
+        Show this link for courses where financial assistance is available, unless upgrade deadline has passed
         """
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+        if datetime.datetime.now(pytz.UTC) >= enrollment.course_upgrade_deadline:
```

Comment:
If the upgrade deadline is equal to today's date, it will not show the code? Should we ask product how many days it makes sense to show the link to (let's say there's one day left) does it still make sense to add it?

Reply:
Good call!  I updated it now to be only > (not >=) 
Shelby confirmed that hiding after it's expired is enough, fine to still show link on the deadline day